### PR TITLE
[nodejs client] fix hookup issues in src/core/server/opensearch

### PR DESCRIPTION
**Pull request by ananzh**
### Description
1. import Client from '@opensearch-project/opensearch/api/new' to hook
up the new types
2. remove asynSearch
3. apply type assertion
4. fix undefined and mismatched types

### Issues Resolved

Resolves https://github.com/opensearch-project/OpenSearch-Dashboards/issues/1193

Partially addresses https://github.com/opensearch-project/OpenSearch-Dashboards/issues/837

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [ ] Commits are signed per the DCO using --signoff

### CODE REVIEW

**Comment by kavilla** [comment]
> remove asyncSearch

Looks asyncSearch was only really used by `x-pack` plugins correct? So that's why it's safe to remove (especially if only in tests).

**Comment by ananzh** [comment]
> > remove asyncSearch
> 
> Looks asyncSearch was only really used by `x-pack` plugins correct? So that's why it's safe to remove (especially if only in tests).

yeah from my search, it is still a x-pack feature in 7.10

File: src/core/server/opensearch/version_check/ensure_opensearch_version.test.ts [hunk 1]
```diff
@@ -169,7 +169,7 @@ describe('pollOpenSearchNodesVersion', () => {
     internalClient.nodes.info.mockImplementationOnce(() => createOpenSearchError(error));
   };
 
-  it('returns iscCompatible=false and keeps polling when a poll request throws', (done) => {
```
Comment (kavilla) [inline src/core/server/opensearch/version_check/ensure_opensearch_version.test.ts:172]: nice catch

**Review by kavilla — APPROVED** [review]

File: src/core/server/opensearch/client/retry_call_cluster.test.ts [hunk 2]
```diff
@@ -62,13 +62,13 @@ describe('retryCallCluster', () => {
       ...dummyBody,
     });
 
-    client.asyncSearch.get
```
Comment (kavilla) [inline src/core/server/opensearch/client/retry_call_cluster.test.ts:65]: NIT: It's just a test file but it seems like we removed asyncSearch but these tests are made for asyncSearch. Do we need them anymore?

Comment (tmarkley) [reply src/core/server/opensearch/client/retry_call_cluster.test.ts:65]: Yeah I'd say if we're testing `search` elsewhere these can be removed.

**Review by tmarkley — APPROVED** [review]

**Comment by kavilla** [comment]
Can we merge this?

**Pull request merged**